common/mqtt_client/tests/integration_test.rs

```diff
@@ -27,7 +27,7 @@ fn sending_and_receiving_a_message() {
 
     let payload = String::from("Hello there!");
     match tokio_test::block_on(scenario(payload.clone())) {
-        Ok(Some(rcv_message)) => assert_eq!(rcv_message.payload_trimmed(), payload.as_bytes()),
+        Ok(Some(rcv_message)) => assert_eq!(rcv_message.payload_str().unwrap(), payload),
```

Comment:
Is it a right approach to use `payload_str()` instead of making `payload_trimmed()` public?

Reply:
Yes, because now  payload_trimmed is a private function. Now the payload_str will be the interface and works on strings directly instead of utf8 bytes.